### ISSUE TITLE
Replace zoom limits with fewer details endpoint

### ISSUE THREAD

**Issue by RyanTG** [pinballmap/pbm-react#563]
### Overall goal:

Get rid of "zoom in" limit. Replace with a `within_bounding_box?no_details=2` endpoint that returns fewer location/machine info, has no marker labels, and uses simpler marker styles.

### Zoom thresholds

I think our current zoom thresholds (which are different depending on the filters applied) can be used as is. When a threshold is exceeded, instead of preventing refreshes, we switch the endpoint flag from `no_details=1` to `no_details=2`.

### no_details=2 

The flag will only return:
- location name
- location city
- location lat
- location lon

When this flag is used, we should hide:
- The `List` button
- The "filter by machines shown on the map" option.

Marker clicks can still show the bottom sheet, but the bottom sheet will only contain the location name and city, and maybe text that says "click for more details"

### Map markers

The current CustomMapMarker component contains a lot of things that wouldn't be used when this flag is used. 

Such as:
- No labels (floating location names or number of machines).
- Sort order is not necessary.

The marker icon will be a small simple white circle png. I'm still unsure if we'll show saved location map markers, but I'm leaning toward not so that we don't have to bother with sorting those above other markers. I guess it would still be good to have a "selected" marker.

The `iconSize: ["interpolate", ["linear"], ["zoom"]` will be different for these markers.

All this is toe say that it may be worth creating a separate CustomMapMarker2 component or something that is used in this case.

**Comment by RyanTG** [comment]
One potential alternative to _only_ using Zoom thresholds to switch to the no_details=2 endpoint is to use a combo of zoom and the actual number of results. I think this would involve making two queries. 

For example, say that you filter to a machine that isn't common, instead of switching to no_details=2 as you zoom out and showing the simplified info, you switch to no_details=2 in order to quickly check the quantity of locations in the request, and if it's lower than a certain threshold (300?), you then make a no_details=1 request for the same bounds and display that.

**Comment by bpoore** [comment]
@RyanTG, would the dual request option only kick in when we currently would be showing "Please zoom in"? I think we could make that work.

**Comment by RyanTG** [comment]
Yes, that sounds right.

**Comment by RyanTG** [comment]
Commit for this feature on the website: https://github.com/pinballmap/pbm/commit/bf6330d6f7f20eac8ab2a0980fad3257875fc37c#diff-08ce13951cddb70864cea049afd986d569303b434b05202957b5d9ed2b012e98

Lessons learned from the website (basically here's what we did), with ideas for app:

**Requests:** 

_Website:_
Every single request is split into two queries: **one** is a lean query (basically `within_bounding_box?no_details=2`) that we use to construct the geojson for the map data, and then a **second one** (basically `within_bounding_box?no_details=1`) that is used for the results/list AND is limited to returning only 100 items. On the website, the "list" divs gets loaded next to the map, and they take up a lot of time to load. In the app, it seems like this is smoother because the "list" is on a separate screen and doesn't slow the map screen loading down. 

_App:_ 
For the map itself, all we ever need to load map results (including filtered results) is a `?no_details=2` queries that returns only id/lat/lon/machine_count.

But! For the "List" and the "Machines in Map Area" filter, we need more details. For List, it's not reasonable to show thousands of locations, but I also don't think it's worth completely hiding the list (as I previously suggested). Perhaps we follow the website and just add a limit to the number of items returned.

On the website, we sum the number of locations (and machines) from the initial query, and then use that to say "too many locations to show in this list" when the number of locations is greater than 100.

For "Machines in Map Area" I am less certain what to do. We could use the same "too many to list" logic to them hide this filter option so that we don't to worry about joining it with the list of machines.

In summary:  I think it might be ideal to completely do away with all "zoom in" logic and perhaps just do two queries at all times - `?no_details=2` and ``?no_details=1 limited to X`

**Map Display**

_Website:_
Created 9 markers. They vary in color and size.

10+ machines zoomed in 
+ zoomed out
10+ machines selected zoomed in 
2-9 machines zoomed in 
+ zoomed out
1 machine zoomed in
+ zoomed out
1-9 machines selected zoomed in (got lazy and just made one for both of those cases)
selected marker zoomed out (just one for all).

_App:_
Could do the same thing, though in the app we also have Saved markers.

_Website:_ 
Use mapbox/maplibre expressions to switch back and forth between zoomed in/out markers as user zooms. And expressions to hide/show labels.

_App:_
Could do same thing.

**Comment by RyanTG** [comment]
Noting before I forget: with pagination in the list screen, the sort buttons would only sort the currently loaded results and not results from additional pages. I am thinking that one potential solution could be to sort in the API query. So switching the sort would make a new query.

**Comment by RyanTG** [comment]
https://github.com/pinballmap/pbm-react/commit/4462684efba0820cb6408799be8b4f787b460106

One data request for map markers. geojson format and very lean.

Another data request when LocationList is visited, and list is paginated.

Made a new endpoint param for the Machines in Map Area list that is only requested when FindMachine is visited.

Made a new endpoint param for including distance in within_bounding_box so that we can sort by Near.

Minor todo is to review all assets to make sure they are as small as can be.
